### ISSUE TITLE
Standardize the labels in our GitHub repos

### ISSUE THREAD

**Issue by chasenlehara** [donejs/developer-relations#17]
It’d be great if we consistently used the same labels across all of our GitHub repos.

Here are the ones that GitHub includes with new repos:

- bug
- duplicate
- enhancement
- help wanted
- invalid
- question
- wontfix

Here’s what we regularly use as well:

- design
- developer relations
- documentation
- easy
- p0, p1, p2, p3, & p4
- regression

We can use [github-label-sync](https://www.npmjs.com/package/github-label-sync) to accomplish this goal. Ideally, this repo would contain a script that could be run whenever we needed to re-synchronize our labels.

**Comment by chasenlehara** [comment]
I think https://github.com/bitovi/brain/issues/32 needs to be done before we change Waffle’s “ready” labels to “help wanted”

**Comment by chasenlehara** [comment]
For reference, here’s a list of all the labels from all of our repos in ZenHub:

- 4.0
- approved
- Breaking Change
- Bug
- bug
- compatibility
- Compatibility
- design
- Dev Build
- developer relations
- developer relations & evangelism
- Difficulty - Average
- Difficulty - Hard
- discussion
- Discussion
- Documentation
- documentation
- duplicate
- Easy
- easy
- enhancement
- Epic
- error messages
- error messaging
- Error Messaging
- essential
- feature
- Fixed In Branch
- fixed in branch
- funcunit.com
- Future Feature
- greenkeeper
- guide
- hackillinois
- help wanted
- in major
- in progress
- In Progress
- in-progress
- invalid
- live-reload
- meetup content
- need more info
- Needs a hero
- needs help
- Needs Review
- npm
- P0
- P1
- P2
- P3
- P4
- provider
- question
- ready
- ready for review
- review
- revisit
- sample code
- solved on paper
- Solved on paper
- sourcemaps
- spring cleaning
- styles
- task
- Tech Debt
- tests
- training content
- upstream
- UX
- website
- windows

**Comment by chasenlehara** [comment]
Here’s a full list of labels that I think we should standardize, with aliases included that need to be renamed:

- bug: Bug
- cleanup: spring cleaning, Tech Debt
- compatibility: Compatibility
- design: styles, UX
- developer relations: developer relations & evangelism
- discussion: Discussion
- documentation: Documentation
- duplicate
- easy: Easy
- enhancement
- error messages: error messaging, Error Messaging
- fixed in branch: Fixed In Branch
- help wanted: ready
- in progress: In Progress, in-progress
- invalid
- needs a hero: Needs a hero
- p0: P0
- p1: P1
- p2: P2
- p3: P3
- p4: P4
- regression
- question
- solved on paper: Solved on paper
- website: funcunit.com
- wontfix

**Comment by chasenlehara** [comment]
I made a script that can sync the labels for just a single repo or an entire org: https://github.com/donejs/developer-relations/blob/c6c4769a82895890c8545a1bbd86595130810bd0/sync-labels.js

These are all the labels it adds to each repo (if they aren’t already there). It’ll rename existing labels if it recognizes them as an aliases. https://github.com/donejs/developer-relations/blob/c6c4769a82895890c8545a1bbd86595130810bd0/labels.json

If you run the script on an entire org, it’ll get a list of all the repos, then prompt you before it changes each one. It shows the changes it’ll make so it’s clear what’s about to happen.

@justinbmeyer, @matthewp, @phillipskevin, I’ve already run this on the `bit-docs` org ([example](https://github.com/bit-docs/bit-docs/labels)). I’m going to run it on the CanJS, DoneJS, and StealJS orgs as early as tomorrow as long as there are no objections.

**Comment by chasenlehara** [comment]
Looks like we’re affected by https://github.com/Financial-Times/github-label-sync/issues/4. When the script is run multiple times on https://github.com/canjs/canjs, it prompts to add some later labels (solved on paper, website, etc.) that already exist.

**Comment by chasenlehara** [comment]
I finished standardizing the labels across all of our orgs.
